src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php

```diff
@@ -123,43 +123,57 @@ public function databaseRefreshLando(string $siteName = 'default'): Result
     /**
      * Refresh database on Tugboat.
      */
-    public function databaseRefreshTugboat(): Result
+    public function databaseRefreshTugboat(): ResultData
     {
         $this->io()->title('refresh tugboat databases.');
-        $result = null;
+        $resultData = new ResultData();
+
         foreach (array_keys($this->getAllSitesConfig()) as $siteName) {
+            $dbPath = '';
             try {
                 $dbPath = $this->databaseDownload($siteName);
-            } catch (TaskException) {
+            } catch (TaskException $e) {
                 $this->yell("$siteName: No database configured. Download/import skipped.");
+                $resultData->append($e->getMessage());
                 // @todo: Should we run a site-install by default?
                 continue;
             }
-            if (!is_string($dbPath) || strlen($dbPath) == 0) {
+            if (!is_string($dbPath) || $dbPath === '') {
                 $this->yell("'$siteName' database path not found.");
+                $resultData->append("'$siteName' database path not found.");
                 continue;
             }
             $dbName = $siteName === 'default' ? 'tugboat' : $siteName;
-            $result = $this->taskExec('mysql')
+            $taskResult = $this->task(Alternatives::class, 'mariadb', 'mysql')->run();
+            if (!$taskResult->wasSuccessful()) {
+                $resultData->append($taskResult);
+                continue;
+            }
+            $dbDriver = $taskResult->getData()['path'];
```

Comment:
There is only one instance of `$taskResult->getData()['path'];` in this PR, but I am imagining a future where this is used more commonly throughout the codebase. Would it be even better to have our method call to the `Alternatives` class return the path directly without having the caller need to know what to do with it and what array key to look for?

Reply:
Looking into it more, the problem is that `getData()` contract is supposed to return an array. Setting it to a string in a custom Result class would break the interface. 

So I could offer both: A convenience method for users who want one, and the regular route for people already accustomed to Robo and who will just fish for their data using `getData()` because that's what they are used to.

Reply:
@apotek You've laid it all out very well. What do you think we should do??

Reply:
> @apotek You've laid it all out very well. What do you think we should do??

I messed around with decorators and, well of course even a decorator can't override an interface. A contract is a contract, so an array must be returned by getData(). 

That leaves option 2, which is putting a helper function in the Task itself rather than the Result object. But the pattern of Tasks is that they return a Result object and you work with that Result object, not that you run the task, ignore the Result (or sniff it) and then call some other method on the Task to get a specific kind of result. So, I _can_ implement a convenience method, but I feel that it goes against a pattern, is therefore unexpected, and would force a developer to study a README rather than do what they do with any other Robo task, which is look at the returned Result data via getData() and figure it out from there. 

So in the end, my answer is to leave it the way it is. ☘️